src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,80 @@
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final CategoryService categoryService;
+    private final ProductMapper productMapper;
+
+    public ProductController(ProductService productService,
+                             CategoryService categoryService,
+                             ProductMapper productMapper) {
+        this.productService = productService;
+        this.categoryService = categoryService;
+        this.productMapper = productMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto save(@RequestBody ProductRequestDto requestDto) {
+        Product product = productMapper.toModel(requestDto);
+        return productMapper.toDto(productService.save(product));
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto findById(@PathVariable Long id) {
+        return productMapper.toDto(productService.findById(id));
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto requestDto) {
+        Product product = productMapper.toModel(requestDto);
+        product.setId(id);
+        return productMapper.toDto(productService.update(product));
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @GetMapping("/price")
+    public List<ProductResponseDto> findAllByPriceBetween(@RequestParam BigDecimal from,
+                                                          @RequestParam BigDecimal to) {
+        List<Product> productsByPrice = productService.findAllByPriceBetween(from, to);
+        return productsByPrice.stream()
+                .map(productMapper::toDto)
+                .toList();
+    }
+
+    @GetMapping("/categories")
+    public List<ProductResponseDto> findAllByCategoryIn(@RequestParam Collection<Long> id) {
+        List<Category> categories = id.stream()
+                .map(categoryService::findById)
+                .toList();
+        return productService.findAllByCategoryIn(categories).stream()
+                .map(productMapper::toDto)
+                .toList();
+    }
```

Comment:
you can try using something  like `List<Product> findAllByCategoryIdIn(List<Long> categoryIds);` repository method also has to work and simplify  your solution